packages/components/src/components/inputs/Input/index.native.tsx

```diff
@@ -1,20 +1,156 @@
+    padding: 0 ${props => (props.hasIcon ? '40px' : '12px')} 0 12px;
+
+    font-size: ${props => (props.isSmallText ? FONT_SIZE_NATIVE.SMALL : FONT_SIZE_NATIVE.BASE)};
+    font-weight: ${FONT_WEIGHT.MEDIUM};
+    color: ${props => (props.color ? props.color : colors.TEXT)};
+
+    border-radius: 2px;
+    
+    ${props =>
+        props.hasAddon &&
+        css`
+            border-top-right-radius: 0;
+            border-bottom-right-radius: 0;
+        `}
+
+    border: 1px solid ${props => props.border};
+
+    background-color: ${colors.WHITE};
+
+    ${props =>
+        props.tooltipAction &&
+        css`
+            z-index: 10001; /* bigger than modal container */
+            position: relative;
+        `};
+`;
+
+const StyledIcon = styled(Icon)`
+    position: absolute;
+    top: 12;
+    right: 15;
+    z-index: 10001;
+`;
+
+const BottomText = styled.Text<InputProps>`
+    margin-top: 10px;
+    font-size: ${FONT_SIZE_NATIVE.SMALL};
+    color: ${props => (props.color ? props.color : colors.TEXT_SECONDARY)};
+    z-index: 0;
+`;
+
+const Overlay = styled.View<InputProps>`
+    ${props =>
+        props.isPartiallyHidden &&
+        css`
+            bottom: 0;
+            border: 1px solid ${colors.DIVIDER};
+            border-radius: 2px;
+            position: absolute;
+            width: 100%;
+            height: 100%;
+            background-image: linear-gradient(
+                to right,
+                rgba(0, 0, 0, 0) 0%,
+                rgba(249, 249, 249, 1) 220px
+            );
+        `}
+`;
+
+const TooltipAction = styled.View`
+    display: ${(props: { action?: React.ReactNode }) => (props.action ? 'flex' : 'none')};
+    margin: 0px 10px;
+    position: absolute;
+    top: 5;
+    background: black;
+    height: 37;
+    border-radius: 5px;
+    z-index: 10009;
+    transform: translate(-1px, -1px);
+`;
+
+const TooltipActionText = styled.Text`
+    color: ${colors.WHITE};
+    padding: 0 14px;
+    height: 37;
+    line-height: ${LINE_HEIGHT.TREZOR_ACTION};
+`;
+
+const ArrowUp = styled.View`
+    position: absolute;
+    top: -9;
+    left: 12;
+    width: 0;
+    height: 0;
+    background-color: transparent;
+    border-style: solid;
+    border-left-width: 9;
+    border-right-width: 9;
+    border-bottom-width: 9;
+    border-left-color: transparent;
+    border-right-color: transparent;
+    border-bottom-color: black;
+`;
+
+interface WrapperProps {
+    className?: string;
+}
+
+// TODO: fix input props without omit
+interface InputProps
+    extends Omit<
+        React.InputHTMLAttributes<HTMLInputElement>,
+        | 'style'
+        | 'onFocus'
+        | 'onBlur'
+        | 'onChange'
+        | 'onKeyPress'
+        | 'onTouchCancel'
+        | 'onTouchEnd'
+        | 'onTouchEndCapture'
+        | 'onTouchMove'
+        | 'onTouchStart'
+        | 'onScroll'
+        | 'defaultValue'
+        | 'autoCapitalize'
+        | 'autoCorrect'
+        | 'value'
+    > {
```

Comment:
that looks nasty :D. imo the problem is that we need to extend some react-native element (Idk how or if it is actually doable), not html input element.

`import { Text, TextProperties } from 'react-native';`
`TextProperties` looks promising.

Reply:
fixed https://github.com/trezor/trezor-suite/pull/272/commits/7373671c743a5e9273423cb07933d98302daa338

Reply:
nice!